ostd/src/task/processor.rs

```diff
@@ -90,6 +92,11 @@ pub(super) fn switch_to_task(next_task: Arc<Task>) {
     //    context and the next task context.
     // 2. The next task context is a valid task context.
     unsafe {
+        let fsbase = (*next_task_ctx_ptr).fsbase;
+        if fsbase != 0 {
+            wrfsbase(fsbase as u64);
```

Comment:
Here we should have an abstraction in the architecture dependent module for writing the TLS pointer. This is an architecture-independent module.